### PR TITLE
Javascript Dates manipulation with date-fns

### LINKED ISSUE #326: Javascript Date manipulation with date-fns
### Brief Summary: 
At times during the development process, we constantly jump into date objects and we need tools to assist us to handle those instances. Date-fns is a lightweight date library that provides comprehensive functions toolset for date formating and manipulation with a lot of simplicity. There are two big players (MomentJS and date-fns) when it comes to javascript date management. This article will comprehensively cover date-fns with the following key takeaways:
### Key Takeaways: 
•	What is date-fns
•	why date-fns
•	Getting started with date-fns
•	Date values parsing and formating
•	Date arithmetics( additions, substractions)
•	Date locale
•	Date comparisons
•	Date timezones
•	Date validation
•	Differences between dates
•	Comparison between moment and date-fns
•	Other date manipulating framework alternatives
### References: 
https://date-fns.org/docs/Getting-Started

### PR DESCRIPTION
**Pull request by kimkimani**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? see the example [Formats here](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)
- [x] Is your article properly formatted in Markdown? - see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
- [x] Have you used the correct folder and file structure? - see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)
- [x] Is your article over 750 words?
- [x] Have you used our preferred heading structure? - [H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)
- [x] Does your article provide enough value and detail about your topic? Articles should be clear, accurate, and fully explained.
- [x] Can your article be understood by beginners? **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image? (600X400 pixels (images can be resized in paint) - also please make sure the image is from a site like www.unsplash.com)
- [x] In your PR - add a comment that states ["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278) to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - Tool: [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - Tool: [Hemmingway](http://www.hemingwayapp.com/)
- [x] Have you added sources for quotes and images that aren't yours?

### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting/)
- [x] Have you used Repl.it for your code snippets?

### Contribution Guidelines
For first-time contributors and for more details, see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)
This fixes #326

### CODE REVIEW

**Comment by kimkimani** [comment]
"This fixes #326 #

**Comment by LinusMuema** [comment]
Hi @kimkimani 
This is a great article. Very informative 🚀. I have noticed that your sentences tend to be a bit long. You can use a free tool recommended called [Heningway](http://www.hemingwayapp.com/) to check your sentences and improve your article grammar.

File: articles/javascript-dates-manipulation-with-date-fns/index.md [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+console.log(`${format(date, 'EEEE,MMMM do, yyyy hh:mm a')}`);
+console.log(`${format(date, 'MMMM, yyyy')}`);
+console.log(`${format(date, 'MMMM.do.')}`);
+console.log(`${format(date, 'EEEE do HH:mm ')}`);
+console.log(`${format(date, 'EEEE,MMMM do, yyyy ppppp')}`);
+console.log(`${format(date, 'do  MMMM yyyy OOOO')}`);
+```
+The above examples will display the date parsed with several date formats. [Check out more date formats](https://date-fns.org/v2.16.1/docs/format) you can play with and get more ideas of how date-fns formats works.
+
+**Note:** we have only imported the `format` function as it is what we need for the moment, that's one of the dynamic features of date-fns. When formatting date-fns token values try to avoid some [common mistakes](https://github.com/date-fns/date-fns/blob/master/docs/unicodeTokens.md) such as:
```
Comment (LinusMuema) [inline articles/javascript-dates-manipulation-with-date-fns/index.md:121]: I would suggest you give a brief description of some of the common mistakes and how one can avoid them. For example how one can confuse the `dd` and `DD`.

File: articles/javascript-dates-manipulation-with-date-fns/index.md [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+Syntax example:
+`subDays(date, amount)`
+Import sub function as shown in the example below. Similarly, you can choose format function to manipulate your display options.
+```js
+const date = new Date('2020.09.29 10:12:00');
+//sub days
+const a = subDays(date, 4);
+//sub months
+const b = subMonths(date, 3);
+//sub years
```
Comment (LinusMuema) [inline articles/javascript-dates-manipulation-with-date-fns/index.md:157]: Maybe this is more of a technical issue but I think there is a missing line of code. The `sub years` part.

**Comment by kimkimani** [comment]
@LinusMuema should be better now

**Review by LinusMuema — COMMENTED** [review]
Awesome work in shortening some of the sentences.

**Comment by LinusMuema** [comment]
Hi @hectorkambow 
I have done a technical review and all the code snippets run successfully and the repl.it snippets too.
You can do a final review and see what else needs to be done.

@kimkimani I would suggest you have a consistent way of presenting your code. You can choose either the code snippets or the repl.it snippets. This ensures uniformity and readers have a smooth flow when reading your article. The repl.it can work better.

**Comment by hectorkambow** [comment]
@kimkimani ? about the hero image - did you make this hero image? or did you grab it online?

**Comment by hectorkambow** [comment]
plagiarism check is done

**Comment by kimkimani** [comment]
yeah @hectorkambow

**Comment by kimkimani** [comment]
> @kimkimani ? about the hero image - did you make this hero image? or did you grab it online?

yeah @hectorkambow

**Comment by kimkimani** [comment]
> > @kimkimani ? about the hero image - did you make this hero image? or did you grab it online?
> 
> yeah @hectorkambow

I got it online

**Comment by hectorkambow** [comment]
👋  Good morning @mwickman Matt - hope you're having a great start to the week! 
Wanted to ask if you could give this "Javascript Dates manipulation with date-fns" article a quick tech review - It has been reviewed by one of our newest Peer Reviewers and (should be a lot smoother) I'm trying to double-check the tech accuracy before we publish. 
All in an effort to help educate our newer Peer Reviewers the revisions process and what they should be looking for moving forward. 
Thank you 🚀

**Review by mwickman — APPROVED** [review]
Nice thorough comparison showing the landscape of Date libs. Love the many usage examples! Article looks good to me.

**Pull request merged**